src/xpu/sycl/types.hpp

```diff
@@ -93,6 +93,37 @@ using in_memory_arg_t = memory_arg_t<::sycl::access::mode::read>;
 using out_memory_arg_t = memory_arg_t<::sycl::access::mode::write>;
 using inout_memory_arg_t = memory_arg_t<::sycl::access::mode::read_write>;
 
+//TODO: This is a work-around for reducing the size of kernel parameters being passed
+// to the matmul kernel. This is to be removed when we shift to sycl-RTC
+struct md_t_spec_const {
+    static constexpr int max_dims = 6;
+
+    using dim32_t = int32_t;
+    using dims32_t = dim32_t[max_dims];
+
+    // ordering of elements is important during initialization.
+    // This struct cannot have a non trivial constructor, or any non trivial types.
+    data_type_t data_type_;
+
+    dim32_t ndims_;
+
+    dims32_t dims_;
+    dims32_t padded_dims_;
+    dims32_t padded_offsets_;
+    dim32_t offset0_;
+
+    dims32_t strides_;
+    dim32_t inner_nblks_;
+    dims32_t inner_blks_;
+    dims32_t inner_idxs_;
+};
```

Comment:
// This struct cannot have a non trivial constructor, or any non trivial types, because...
// why exactly?
`static_assert(std::is_trivial_v<md_t_spec_const>, "md_t_spec_const must be trivial");`

Reply:
From the sycl specification [HERE](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_declaring_a_specialization_constant) we need the spec constant to be `device_copyable` [HERE](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec::device.copyable).

I can clarify this in the comment if you think that makes sense, or remove it all together?

Reply:
Looking into this a bit more DPC++ defines `SYCL_DEVICE_COPYABLE=1` so we may be able to avoid the need of `md_t_spec_const` if we specialize `is_device_copyable_v<md_t>` to return `true`.

Reply:
All the knowledge not available from reading the code better be put in a comment as a reference with exact spot to look or as a summary.